engine/src/main/java/com/google/android/fhir/db/impl/dao/LocalChangeDao.kt

```diff
@@ -123,9 +180,52 @@ internal abstract class LocalChangeDao {
         payload = "",
         versionId = remoteVersionId,
       ),
+      emptyList(),
     )
   }
 
+  private fun extractResourceReferences(resource: Resource) =
+    fhirTerser.getAllResourceReferences(resource).toSet()
+
+  /**
+   * Extract the difference in the [ResourceReferenceInfo] by getting all the references from both
+   * the versions of the resource and then finding out the difference in the two sets.
+   *
+   * Two versions of a resource can vary in two ways in terms of the resources they refer:
+   * 1) A reference present in resource1 is removed and is not present in resource2. Such
+   *    differences in resource references can be extracted by subtracting the set of references in
+   *    the resource2 from the set of references in the resource1
+   * 2) A new reference is added to the resource1. This implies that the reference is present in
+   *    resource2 and not in resource1. Such differences in resource references can be extracted by
+   *    subtracting the set of references in the resource1 from the set of references in the
+   *    resource2
+   *
+   * Combining the above two types of differences would give the entire set of difference in the two
+   * versions of the resource.
+   *
+   * This method is useful to extract differences for UPDATE kind of [LocalChange]
+   */
```

Comment:
The kdoc seems verbose and repetitive in some places. To enhance clarity and conciseness, I would recommend the following improvements:
```suggestion
/**
 * Extracts the difference in [ResourceReferenceInfo] between two versions of a resource.
 *
 * A difference between two resource versions, in terms of references, can manifest in two ways:
 * 1) A reference is present in `resource1` but not in `resource2`.
 * 2) A reference is present in `resource2` but not in `resource1`.
 *
 * This method computes both types of differences and returns the combined set of differences. 
 * It is particularly useful for extracting differences during an UPDATE operation on a [LocalChange].
 *
 * @param resource1 The first version of the resource.
 * @param resource2 The second version of the resource.
 * @return A set of [ResourceReferenceInfo] indicating the differences in references between the two resource versions.
 */
```